### ISSUE TITLE
Traceback when re-adding a previously deleted channel

### ISSUE THREAD

**Issue by pllim** [ejeschke/ginga#242]
You can reproduce the following traceback in the latest dev version of Ginga with the following workflow:

1. Start Ginga.
2. Click `Channel -> Add Channel` to create "Image0" channel.
3. Delete "Image0" channel.
4. Repeat step 2.

Traceback:
```python
2015-12-08 15:39:16,517 | E | Callback.py:88 (make_callback) | Error making callback 'set': wrapped C/C++ object of type QLabel has been deleted
2015-12-08 15:39:16,517 | E | Callback.py:89 (make_callback) | Traceback:
  File ".../ginga/misc/Callback.py", line 72, in make_callback
    res = method(*cb_args, **cb_kwdargs)

  File ".../ginga/misc/plugins/Info.py", line 193, in zoomset_cb
    info.winfo.zoom.set_text(text)

  File ".../ginga/qtw/Widgets.py", line 238, in set_text
    self.widget.setText(text)

2015-12-08 15:39:16,523 | E | Callback.py:88 (make_callback) | Error making callback 'set': wrapped C/C++ object of type QLabel has been deleted
2015-12-08 15:39:16,526 | E | Callback.py:89 (make_callback) | Traceback:
  File ".../ginga/misc/Callback.py", line 72, in make_callback
    res = method(*cb_args, **cb_kwdargs)

  File ".../ginga/misc/plugins/Info.py", line 206, in autozoom_cb
    info.winfo.zoom_new.set_text(option)

  File ".../ginga/qtw/Widgets.py", line 238, in set_text
    self.widget.setText(text)

2015-12-08 15:39:16,532 | E | Callback.py:88 (make_callback) | Error making callback 'set': wrapped C/C++ object of type QLabel has been deleted
2015-12-08 15:39:16,535 | E | Callback.py:89 (make_callback) | Traceback:
  File ".../ginga/misc/Callback.py", line 72, in make_callback
    res = method(*cb_args, **cb_kwdargs)

  File ".../ginga/misc/plugins/Info.py", line 198, in cutset_cb
    info.winfo.cut_low_value.set_text('%.4g' % (loval))

  File ".../ginga/qtw/Widgets.py", line 238, in set_text
    self.widget.setText(text)

2015-12-08 15:39:16,547 | E | ImageView.py:996 (redraw_now) | Error redrawing image: wrapped C/C++ object of type RenderWidgetZoom has been deleted
2015-12-08 15:39:16,551 | E | ImageView.py:1001 (redraw_now) | Traceback:
  File ".../ginga/ImageView.py", line 986, in redraw_now
    self.update_image()
  File ".../ginga/qtw/ImageViewQt.py", line 305, in update_image
    self.imgwin.update()

2015-12-08 15:39:16,556 | E | ImageView.py:996 (redraw_now) | Error redrawing image: wrapped C/C++ object of type RenderWidgetZoom has been deleted
2015-12-08 15:39:16,564 | E | ImageView.py:1001 (redraw_now) | Traceback:
  File ".../ginga/ImageView.py", line 986, in redraw_now
    self.update_image()
  File ".../ginga/qtw/ImageViewQt.py", line 305, in update_image
    self.imgwin.update()

2015-12-08 15:39:16,573 | E | Callback.py:88 (make_callback) | Error making callback 'set': wrapped C/C++ object of type QLabel has been deleted
2015-12-08 15:39:16,577 | E | Callback.py:89 (make_callback) | Traceback:
  File ".../ginga/misc/Callback.py", line 72, in make_callback
    res = method(*cb_args, **cb_kwdargs)

  File ".../ginga/misc/plugins/Info.py", line 193, in zoomset_cb
    info.winfo.zoom.set_text(text)

  File ".../ginga/qtw/Widgets.py", line 238, in set_text
    self.widget.setText(text)

2015-12-08 15:39:16,588 | E | ImageView.py:996 (redraw_now) | Error redrawing image: wrapped C/C++ object of type RenderWidgetZoom has been deleted
2015-12-08 15:39:16,592 | E | ImageView.py:1001 (redraw_now) | Traceback:
  File ".../ginga/ImageView.py", line 986, in redraw_now
    self.update_image()
  File ".../ginga/qtw/ImageViewQt.py", line 305, in update_image
    self.imgwin.update()

2015-12-08 15:39:16,596 | E | ImageView.py:996 (redraw_now) | Error redrawing image: wrapped C/C++ object of type RenderWidgetZoom has been deleted
2015-12-08 15:39:16,598 | E | ImageView.py:1001 (redraw_now) | Traceback:
  File ".../ginga/ImageView.py", line 986, in redraw_now
    self.update_image()
  File ".../ginga/qtw/ImageViewQt.py", line 305, in update_image
    self.imgwin.update()

2015-12-08 15:39:16,611 | E | ImageView.py:996 (redraw_now) | Error redrawing image: wrapped C/C++ object of type RenderWidgetZoom has been deleted
2015-12-08 15:39:16,616 | E | ImageView.py:1001 (redraw_now) | Traceback:
  File ".../ginga/ImageView.py", line 986, in redraw_now
    self.update_image()
  File ".../ginga/qtw/ImageViewQt.py", line 305, in update_image
    self.imgwin.update()

2015-12-08 15:39:16,620 | E | ImageView.py:996 (redraw_now) | Error redrawing image: wrapped C/C++ object of type RenderWidgetZoom has been deleted
2015-12-08 15:39:16,624 | E | ImageView.py:1001 (redraw_now) | Traceback:
  File ".../ginga/ImageView.py", line 986, in redraw_now
    self.update_image()
  File ".../ginga/qtw/ImageViewQt.py", line 305, in update_image
    self.imgwin.update()

2015-12-08 15:39:16,628 | E | Callback.py:88 (make_callback) | Error making callback 'set': wrapped C/C++ object of type QLabel has been deleted
2015-12-08 15:39:16,631 | E | Callback.py:89 (make_callback) | Traceback:
  File ".../ginga/misc/Callback.py", line 72, in make_callback
    res = method(*cb_args, **cb_kwdargs)

  File ".../ginga/misc/plugins/Info.py", line 213, in autocenter_cb
    info.winfo.center_new.set_text(option)

  File ".../ginga/qtw/Widgets.py", line 238, in set_text
    self.widget.setText(text)

2015-12-08 15:39:16,634 | E | Callback.py:88 (make_callback) | Error making callback 'set': wrapped C/C++ object of type QLabel has been deleted
2015-12-08 15:39:16,637 | E | Callback.py:89 (make_callback) | Traceback:
  File ".../ginga/misc/Callback.py", line 72, in make_callback
    res = method(*cb_args, **cb_kwdargs)

  File ".../ginga/misc/plugins/Info.py", line 193, in zoomset_cb
    info.winfo.zoom.set_text(text)

  File ".../ginga/qtw/Widgets.py", line 238, in set_text
    self.widget.setText(text)

2015-12-08 15:39:16,643 | E | Callback.py:88 (make_callback) | Error making callback 'set': wrapped C/C++ object of type QLabel has been deleted
2015-12-08 15:39:16,648 | E | Callback.py:89 (make_callback) | Traceback:
  File ".../ginga/misc/Callback.py", line 72, in make_callback
    res = method(*cb_args, **cb_kwdargs)

  File ".../ginga/misc/plugins/Info.py", line 193, in zoomset_cb
    info.winfo.zoom.set_text(text)

  File ".../ginga/qtw/Widgets.py", line 238, in set_text
    self.widget.setText(text)

2015-12-08 15:39:16,653 | E | Callback.py:88 (make_callback) | Error making callback 'set': wrapped C/C++ object of type QLabel has been deleted
2015-12-08 15:39:16,656 | E | Callback.py:89 (make_callback) | Traceback:
  File ".../ginga/misc/Callback.py", line 72, in make_callback
    res = method(*cb_args, **cb_kwdargs)

  File ".../ginga/misc/plugins/Info.py", line 193, in zoomset_cb
    info.winfo.zoom.set_text(text)

  File ".../ginga/qtw/Widgets.py", line 238, in set_text
    self.widget.setText(text)

2015-12-08 15:39:16,667 | E | ImageView.py:996 (redraw_now) | Error redrawing image: wrapped C/C++ object of type RenderWidgetZoom has been deleted
2015-12-08 15:39:16,676 | E | ImageView.py:1001 (redraw_now) | Traceback:
  File ".../ginga/ImageView.py", line 986, in redraw_now
    self.update_image()
  File ".../ginga/qtw/ImageViewQt.py", line 305, in update_image
    self.imgwin.update()

2015-12-08 15:39:16,681 | E | ImageView.py:996 (redraw_now) | Error redrawing image: wrapped C/C++ object of type RenderWidgetZoom has been deleted
2015-12-08 15:39:16,687 | E | ImageView.py:1001 (redraw_now) | Traceback:
  File ".../ginga/ImageView.py", line 986, in redraw_now
    self.update_image()
  File ".../ginga/qtw/ImageViewQt.py", line 305, in update_image
    self.imgwin.update()

2015-12-08 15:39:16,690 | E | Callback.py:88 (make_callback) | Error making callback 'set': wrapped C/C++ object of type QLabel has been deleted
2015-12-08 15:39:16,692 | E | Callback.py:89 (make_callback) | Traceback:
  File ".../ginga/misc/Callback.py", line 72, in make_callback
    res = method(*cb_args, **cb_kwdargs)

  File ".../ginga/misc/plugins/Info.py", line 203, in autocuts_cb
    info.winfo.cut_new.set_text(option)

  File ".../ginga/qtw/Widgets.py", line 238, in set_text
    self.widget.setText(text)

2015-12-08 15:39:16,695 | E | Callback.py:88 (make_callback) | Error making callback 'set': wrapped C/C++ object of type QLabel has been deleted
2015-12-08 15:39:16,697 | E | Callback.py:89 (make_callback) | Traceback:
  File ".../ginga/misc/Callback.py", line 72, in make_callback
    res = method(*cb_args, **cb_kwdargs)

  File ".../ginga/misc/plugins/Info.py", line 206, in autozoom_cb
    info.winfo.zoom_new.set_text(option)

  File ".../ginga/qtw/Widgets.py", line 238, in set_text
    self.widget.setText(text)
```

**Comment by ejeschke** [comment]
@pllim, I also get the traceback.  But it looks like the channel is successfully created, and I can use it.  Is this the case for you as well?  We should fix the traceback, but I want to understand the severity of the issue.

**Comment by ejeschke** [comment]
It looks like this is due to the channel deletion not being implemented in a couple of the plugins, like `Info`, `Pan` and `Zoom`.

**Comment by ejeschke** [comment]
Actually, this was due to the settings not being removed for the channel.  @pllim, there is a fix in commit 2877dde4639f9d30fbe519d786ffab558fc3433e.  Can you verify that this clears up this issue for you?

**Comment by pllim** [comment]
@ejeschke , yes, the error is gone. Thank you!
